nucleus/payara-modules/requesttracing-core/src/main/java/fish/payara/nucleus/requesttracing/store/strategy/ReservoirTraceStorageStrategy.java

```diff
@@ -51,66 +52,42 @@
 public class ReservoirTraceStorageStrategy implements TraceStorageStrategy {
 
     private final Random random;
-    private long counter;
```

Comment:
I believe this `counter` is the code translation of the wiki article text talking about an _unknown large number_. I think the algorithm got misunderstood here. If there was an ever increasing divider for the `probability` (below) the probability would ever decrease over time (with the number of invocations) making it less and less likely an item is kept. Also inverting this does not make sense. That would increase the likelihood to 1 over time. Both are wrong and must be based in an misunderstanding of the algorithm.